File: coresuite-dotnet-standard-vb/generator/TableReport.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports Microsoft.Extensions.Configuration
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.IO
Imports ceTe.DynamicPDF.Text
Imports ceTe.DynamicPDF.PageElements

Namespace coresuite_dotnet_standard_vb.generator
    Public Class TableReport

        Public Shared Sub Run(outputPdfPath As String)
            ' Create a MyDocument and set it's properties
            Dim MyDocument As ceTe.DynamicPDF.Document = New ceTe.DynamicPDF.Document
            MyDocument.Creator = "Table.aspx"
            MyDocument.Author = "ceTe Software"
            MyDocument.Title = "Table Example"

            ' Establises connection to the database
            Dim Connection As SqlConnection = GetOpenDBConn()
            Dim Data As SqlDataReader = GetContactListData(Connection)

            'Create a Table and set it's properties
            Dim MyTable As Table2 = New Table2(0, 0, 512, 676, ceTe.DynamicPDF.Font.Helvetica, 12)
            MyTable.Border.Width = 1
            MyTable.CellDefault.Border.Width = 1
            MyTable.Border.Color = Grayscale.Black
            MyTable.RepeatColumnHeaderCount = 1
            MyTable.RepeatRowHeaderCount = 1

            ' Builds the report
            BuildTable(Data, MyTable)

            Dim CurrentRow As Integer = 0
            Dim CurrentColumn As Integer = 0
            Do
                CurrentRow = CurrentRow + 1
                CurrentColumn = 1
                AddTableToPage(MyDocument, MyTable, CurrentRow, CurrentColumn)
                Dim OverflowTable As Table2 = MyTable.GetOverflowColumns()
                Do
                    CurrentColumn = CurrentColumn + 1
                    AddTableToPage(MyDocument, OverflowTable, CurrentRow, CurrentColumn)
                    OverflowTable = OverflowTable.GetOverflowColumns()
                Loop While Not OverflowTable Is Nothing
                MyTable = MyTable.GetOverflowRows()
            Loop While Not MyTable Is Nothing


            ' Cleans up database connections
            Data.Close()
            Connection.Close()

            ' Outputs the Table Report to the current web MyPage
            MyDocument.Draw(outputPdfPath)
        End Sub

        Public Shared Sub AddTableToPage(ByVal MyDocument As Document, ByVal MyTable As Table2, ByVal CurrentRow As Integer, ByVal CurrentColumn As Integer)
            Dim MyPage As Page = New Page(PageSize.Letter)
            If Not MyTable Is Nothing Then
                MyPage.Elements.Add(MyTable)
            End If
            MyPage.Elements.Add(New Label("(" & CurrentRow & "," & CurrentColumn & ")", 0, MyPage.Dimensions.Body.Height - 12, MyPage.Dimensions.Body.Width, 12, Font.Helvetica, 12, TextAlign.Center))
            MyDocument.Pages.Add(MyPage)
        End Sub

        Public Shared Sub BuildTable(ByVal Data As SqlDataReader, ByVal MyTable As Table2)
            CreateColumns(MyTable)
            CreateRowHeadings(MyTable)

            While Data.Read()
                CreateRow(MyTable, Data)
            End While
        End Sub

        Public Shared Sub CreateRowHeadings(ByVal MyTable As Table2)
            Dim MyRow As Row2 = MyTable.Rows.Add(40, Font.TimesBold, 12, Grayscale.Black, Grayscale.LightGrey)
            MyRow.CellDefault.Align = TextAlign.Center
            MyRow.CellDefault.VAlign = VAlign.Top
            MyRow.Cells.Add("ID")
            MyRow.Cells.Add("Product Name")
            MyRow.Cells.Add("Supplier ID")
            MyRow.Cells.Add("Category ID")
            MyRow.Cells.Add("Quantity Per Unit")
            MyRow.Cells.Add("Unit Price")
            MyRow.Cells.Add("Unit In Stock")
            MyRow.Cells.Add("Units On Order")
            MyRow.Cells.Add("Reorder Level")
            MyRow.Cells.Add("Discontinued")
        End Sub

        Public Shared Sub CreateRow(ByVal MyTable As Table2, ByVal Data As SqlDataReader)
            Dim MyRow As Row2 = MyTable.Rows.Add(20)
            MyRow.Cells.Add(Data.GetInt32(0).ToString(), Font.Helvetica, 12, Grayscale.Black, Grayscale.LightGrey, 1)
            MyRow.Cells.Add(Data.GetString(1))
            MyRow.Cells.Add(Data.GetInt32(2).ToString())
            MyRow.Cells.Add(Data.GetInt32(3).ToString())
            MyRow.Cells.Add(Data.GetString(4))
            MyRow.Cells.Add(Data.GetDecimal(5).ToString("$0.00"))
            MyRow.Cells.Add(Data.GetInt16(6).ToString())
            MyRow.Cells.Add(Data.GetInt16(7).ToString())
            MyRow.Cells.Add(Data.GetInt16(8).ToString())
            MyRow.Cells.Add(Data.GetBoolean(9).ToString())
        End Sub

        Public Shared Sub CreateColumns(ByVal MyTable As Table2)
            MyTable.Columns.Add(25)
            MyTable.Columns.Add(150)
            MyTable.Columns.Add(90)
            MyTable.Columns.Add(90)
            MyTable.Columns.Add(120)
            MyTable.Columns.Add(60)
            MyTable.Columns.Add(90)
            MyTable.Columns.Add(90)
            MyTable.Columns.Add(90)
            MyTable.Columns.Add(90)
        End Sub
        Public Shared Function GetOpenDBConn() As SqlConnection
            ' Creates and opens a database connection
            Dim Connection As SqlConnection = New SqlConnection(Util.DynamicPDFConfiguration.GetConnectionString("NorthWindConnectionString"))
            Connection.Open()
            Return Connection
        End Function

        Public Shared Function GetContactListData(ByVal Connection As SqlConnection) As SqlDataReader
            ' Creates a DataReader for the report
            Dim Command As SqlCommand = Connection.CreateCommand()
            Command.CommandText = "SELECT * FROM Products where (UnitPrice > 15)"
            Dim DataReader As SqlDataReader = Command.ExecuteReader()

            Return DataReader
        End Function
    End Class
End Namespace

File: coresuite-dotnet-standard-cs/generator/TableReport.cs
using ceTe.DynamicPDF;
using ceTe.DynamicPDF.PageElements;
using Microsoft.Extensions.Configuration;
using System.Data.SqlClient;

namespace coresuite_dotnet_standard_cs.generator
{
    public class TableReport
    {
        public static void Run(string outputPdfPath)
        {
            // Create a document and set it's properties
            Document document = new Document
            {
                Creator = "TableReport",
                Author = "ceTe Software",
                Title = "Table Report"
            };

            // Establises connection to the database
            SqlConnection connection = GetOpenDBConn();
            SqlDataReader data = GetContactListData(connection);

            //Create a Table and set it's properties
            Table2 table = new Table2(0, 0, 512, 676, Font.Helvetica, 12);
            table.Border.Width = 1f;
            table.CellDefault.Border.Width = 1f;
            table.Border.Color = Grayscale.Black;
            table.RepeatColumnHeaderCount = 1;
            table.RepeatRowHeaderCount = 1;

            // Builds the report
            BuildTable(data, table);
            int currentRow = 0;
            int currentColumn = 0;
            do
            {
                currentRow++;
                currentColumn = 1;
                AddTableToPage(document, table, currentRow, currentColumn);
                Table2 overflowTable = table.GetOverflowColumns();
                do
                {
                    currentColumn++;
                    AddTableToPage(document, overflowTable, currentRow, currentColumn);
                    overflowTable = overflowTable.GetOverflowColumns();
                } while (overflowTable != null);
                table = table.GetOverflowRows();
            } while (table != null);

            // Cleans up database connections
            data.Close();
            connection.Close();

            // Outputs the SimpleReport to the current web page
            document.Draw(outputPdfPath);
        }

        public static void AddTableToPage(Document document, Table2 table, int currentRow, int currentColumn)
        {
            Page page = new Page(PageSize.Letter);
            if (table != null)
                page.Elements.Add(table);
            page.Elements.Add(new Label("(" + currentRow + "," + currentColumn + ")", 0, page.Dimensions.Body.Height - 12, page.Dimensions.Body.Width, 12, Font.Helvetica, 12, TextAlign.Center));
            document.Pages.Add(page);
        }

        public static void BuildTable(SqlDataReader data, Table2 table)
        {
            CreateColumns(table);
            CreateRowHeadings(table);
            while (data.Read())
                CreateRow(table, data);
        }

        public static void CreateRowHeadings(Table2 table)
        {
            Row2 row = table.Rows.Add(40, Font.TimesBold, 12, RgbColor.Black, RgbColor.LightGrey);
            row.CellDefault.Align = TextAlign.Center;
            row.CellDefault.VAlign = VAlign.Top;
            row.Cells.Add("ID");
            row.Cells.Add("Product Name");
            row.Cells.Add("Supplier ID");
            row.Cells.Add("Category ID");
            row.Cells.Add("Quantity Per Unit");
            row.Cells.Add("Unit Price");
            row.Cells.Add("Unit In Stock");
            row.Cells.Add("Units On Order");
            row.Cells.Add("Reorder Level");
            row.Cells.Add("Discontinued");
        }

        public static void CreateRow(Table2 table, SqlDataReader data)
        {
            Row2 row = table.Rows.Add(20);
            row.Cells.Add(data.GetInt32(0).ToString(), Font.Helvetica, 12, RgbColor.Black, RgbColor.LightGrey, 1);
            row.Cells.Add(data.GetString(1));
            row.Cells.Add(data.GetInt32(2).ToString());
            row.Cells.Add(data.GetInt32(3).ToString());
            row.Cells.Add(data.GetString(4));
            row.Cells.Add(data.GetDecimal(5).ToString("$0.00"));
            row.Cells.Add(data.GetInt16(6).ToString());
            row.Cells.Add(data.GetInt16(7).ToString());
            row.Cells.Add(data.GetInt16(8).ToString());
            row.Cells.Add(data.GetBoolean(9).ToString());
        }

        public static void CreateColumns(Table2 table)
        {
            table.Columns.Add(25);
            table.Columns.Add(150);
            table.Columns.Add(90);
            table.Columns.Add(90);
            table.Columns.Add(120);
            table.Columns.Add(60);
            table.Columns.Add(90);
            table.Columns.Add(90);
            table.Columns.Add(90);
            table.Columns.Add(90);
        }

        public static SqlConnection GetOpenDBConn()
        {
            // Creates and opens a database connection
            SqlConnection connection = new SqlConnection(Util.DynamicPDFConfiguration.GetConnectionString("NorthWindConnectionString"));
            connection.Open();
            return connection;
        }

        public static SqlDataReader GetContactListData(SqlConnection connection)
        {
            // Creates a DataReader for the report
            SqlCommand command = connection.CreateCommand();
            command.CommandText = "SELECT * FROM Products";
            SqlDataReader dataReader = command.ExecuteReader();
            return dataReader;
        }
    }
}
